### DISCUSSION TITLE
No registry entries without FeatureOracleJavaSoft flag?

### DISCUSSION THREAD

**Discussion by HamsterExAstris** [microsoft/openjdk#329] [Q&A] [answered]
Is it correct that if the FeatureOracleJavaSoft flag is not passed to the installer, that no registry entries will be created?

Currently we let Microsoft Visual Studio 2022 install OpenJDK for us. The resulting OpenJDK install gets added to the path, but without other additional registry entries [the code for the Azure Pipelines agent](https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Misc/layoutbin/powershell/Add-JavaCapabilities.ps1) doesn't detect the `java` capability.

I wanted to make sure that I'm not missing any existing out-of-the-box registry entries that the agent code could use to detect Microsoft's OpenJDK install.

**Comment by HamsterExAstris** [comment, marked as answer]
The registry entries can be found in `HKLM:\SOFTWARE\Microsoft\JDK`. There's already [an open issue and PR](https://github.com/microsoft/azure-pipelines-agent/issues/3719) to update the Azure Pipelines agent accordingly.

**Reply by karianna** [reply]
Thanks for the find and upstream fix!
